### PR TITLE
Build Assets List API

### PR DESCRIPTION
**Pull request by vviveksharma**
Signed-off-by: VivekSharma2607 <sharmavishu760@gmail.com>

### :nut_and_bolt: Description: What code changed, and why?
Created a new API for listing assets 

### :athletic_shoe: How to Build and Test the Change
- rebuild components/automate-gateway/
- rebuild components/compliance-service/
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/compliance-service/reporting/relaxting/assets.go [hunk 1]
```diff
@@ -441,3 +441,37 @@ func (backend ES2Backend) getUnCollectedAssets(ctx context.Context, from int32,
...
+        unreachable,err := backend.getUnReachableAssets(ctx , from , size , boolquery , 10)
+        if err != nil {
+            logrus.Errorf("Unable to get the unreachable assets %v" , err)
+            return nil , err
+        }
+        return unreachable , nil
+    } else if asset_type == "collected" || len(asset_type)==0{
+        collected, err := backend.getCollectedAssets(ctx , from , size , filters , boolquery) 
+        if err != nil {
+            logrus.Errorf("Unable to get the collected assets %v" , err)
```
Comment (daveaugustus) [inline components/compliance-service/reporting/relaxting/assets.go:457]:
```suggestion
            logrus.Errorf("Unable to get the collected assets: %v" , err)
```

File: components/compliance-service/reporting/relaxting/assets.go [hunk 2]
```diff
@@ -441,3 +441,37 @@ func (backend ES2Backend) getUnCollectedAssets(ctx context.Context, from int32,
 	filtQuery.Must(getStartTimeAndEndTimeRangeForAsset(filters))
 	return backend.getAssetsList(ctx, from, size, filtQuery)
 }
+func (backend ES2Backend)GetAsset(ctx context.Context, filters map[string][]string , size int32 , from int32 , asset_type string) ([]*reportingapi.Assets, error) {
+    boolquery := backend.getFiltersQueryForAssetFilters(filters)
+    assets := make([]*reportingapi.Assets, 0)
+    if asset_type == "unreachable" {
+        unreachable,err := backend.getUnReachableAssets(ctx , from , size , boolquery , 10)
+        if err != nil {
+            logrus.Errorf("Unable to get the unreachable assets %v" , err)
```
Comment (daveaugustus) [inline components/compliance-service/reporting/relaxting/assets.go:450]:
```suggestion
            logrus.Errorf("Unable to get the unreachable assets: %v" , err)
```
separate comment and error by `:`

**Review by daveaugustus — APPROVED** [review]

**Comment by daveaugustus** [comment]
Rebase the PR. @vviveksharma

File: components/compliance-service/reporting/relaxting/assets.go [hunk 3]
```diff
@@ -441,3 +441,41 @@ func (backend ES2Backend) getUnCollectedAssets(ctx context.Context, from int32,
...
+    if assetType == "unreachable" {
+        unreachable,err := backend.getUnReachableAssets(ctx , from , size , boolquery , 10)
+        if err != nil {
+            logrus.Errorf("Unable to get the unreachable assets: %v" , err)
+            return nil , err
+        }
+        return unreachable , nil
+<<<<<<< HEAD
+    } else if assetType == "collected" || len(assetType)==0 {
+=======
```
Comment (YashviJain01) [inline components/compliance-service/reporting/relaxting/assets.go:456]: @vviveksharma The branches are not rebased properly. Can you please fix.

File: api/interservice/compliance/reporting/reporting.proto [hunk 4]
```diff
@@ -410,9 +411,21 @@ message Collected {
...
 
+message AssetListRequest {
+	repeated ListFilter filters = 1;
+	int32 size = 2;
+	int32 from = 3;
+	string sort = 4;
+	string assets_type = 5;
+}
+
+message AssetList {
```
Comment (YashviJain01) [inline api/interservice/compliance/reporting/reporting.proto:429]: rename it as AssetListResponse

File: components/compliance-service/reporting/relaxting/assets.go [hunk 5]
```diff
@@ -441,3 +441,41 @@ func (backend ES2Backend) getUnCollectedAssets(ctx context.Context, from int32,
 	filtQuery.Must(getStartTimeAndEndTimeRangeForAsset(filters))
 	return backend.getAssetsList(ctx, from, size, filtQuery)
 }
+func (backend ES2Backend)GetAsset(ctx context.Context, filters map[string][]string , size int32 , from int32 , assetType string) ([]*reportingapi.Assets, error) {
+    boolquery := backend.getFiltersQueryForAssetFilters(filters)
+    assets := make([]*reportingapi.Assets, 0)
+    if assetType == "unreachable" {
```
Comment (YashviJain01) [inline components/compliance-service/reporting/relaxting/assets.go:447]: first condition should we of null check. And have a enum in constants file for all the type of asests

**Pull request closed without merging**